src/Admin/ContentItemAdmin.php

```diff
@@ -11,16 +10,20 @@
 use Sonata\AdminBundle\Datagrid\ListMapper;
 use Sonata\AdminBundle\Form\FormMapper;
 use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Zicht\Bundle\PageBundle\Entity\ContentItem;
 use Zicht\Bundle\PageBundle\Type\ContentItemRegionType;
 use Zicht\Bundle\PageBundle\Type\ContentItemTypeType;
 
 /**
  * Admin for content items.
+ *
+ * @template T of ContentItem
+ * @extends AbstractAdmin<T>
```

Comment:
Is deze toevoeging strict noodzakelijk? Wat gaat er dan mis als dit er niet in zit?

Reply:
In Sonata worden er ook dit soort notaties gebruikt, dus daar kunnen we gebruik van maken. Omdat deze Admin weer ge-extend wordt in projecten, heb ik hier een `@template` annotation in geplaatst. Als je dan in bijv. jouw Quote ContentItem Admin weer zet `@extends ContentItemAdmin<Quote>` dan wordt overal in de parent classes `T` vervangen door `Quote`. Zo weten Psalm en PhpStorm dat als jij in jouw QuoteAdmin de method `getSubject()` uit de Sonata AbstractAdmin aanroept, dat deze een `Quote` object returned (want in de `AbstractAdmin::getSubject()` staat `@return T`